### ISSUE TITLE
Setup fails with postgres authentication failed for user "deis"

### ISSUE THREAD

**Issue by jsierles** [deis/deis#1138]
After some issues with etcd, I wiped out etcd's data, got it working properly, and then ran:

```
# make uninstall
# make pull & make run
```
Everything came up except the controller. See the error at the end.

```
Deis components may take a long time to start the first time they are initialized.
Waiting for 1 of 1 deis-routers to start...
fleetctl --strict-host-key-checking=false start -no-block deis-router.1.service;
Triggered job deis-router.1.service start
Waiting for deis-registry to start...
fleetctl --strict-host-key-checking=false start -no-block registry/systemd/deis-registry.service logger/systemd/deis-logger.service cache/systemd/deis-cache.service database/systemd/deis-database.service
^@Triggered job deis-registry.service start
Triggered job deis-logger.service start
Triggered job deis-cache.service start
Triggered job deis-database.service start
Status: activating (start-post)
Waiting for deis-controller to start...
fleetctl --strict-host-key-checking=false start -no-block controller/systemd/*
Triggered job deis-controller.service start
Waiting for deis-builder to start...
fleetctl --strict-host-key-checking=false start -no-block builder/systemd/*
Triggered job deis-builder.service start
^@One or more services failed! Check which services by running 'make status'
You can get detailed output with 'fleetctl status deis-servicename.service'
This usually indicates an error with Deis - please open an issue on GitHub or ask for help in IRC
make: *** [start] Error 1
Ubrique:deis joshua$
Ubrique:deis joshua$ make status
fleetctl --strict-host-key-checking=false list-units
UNIT			STATE		LOAD	ACTIVE	SUB	DESC		MACHINE
deis-builder.service	launched	loaded	active	running	deis-builder	1611bff0.../5.9.96.36
deis-cache.service	launched	loaded	active	running	deis-cache	1611bff0.../5.9.96.36
deis-controller.service	launched	loaded	failed	failed	deis-controller	1611bff0.../5.9.96.36
deis-database.service	launched	loaded	active	running	deis-database	1611bff0.../5.9.96.36
deis-logger.service	launched	loaded	active	running	deis-logger	1611bff0.../5.9.96.36
deis-registry.service	launched	loaded	active	running	deis-registry	1611bff0.../5.9.96.36
deis-router.1.service	launched	loaded	active	running	deis-router	1611bff0.../5.9.96.36
Ubrique:deis joshua$ fleetctl status deis-controller
● deis-controller.service - deis-controller
   Loaded: loaded (/run/fleet/units/deis-controller.service; linked-runtime)
   Active: failed (Result: exit-code) since Fri 2014-06-06 15:45:50 UTC; 10min ago
  Process: 29722 ExecStop=/usr/bin/docker rm -f deis-controller (code=exited, status=0/SUCCESS)
  Process: 28883 ExecStart=/usr/bin/docker run --name deis-controller -p 8000:8000 -e PUBLISH=8000 -e HOST=${COREOS_PRIVATE_IPV4} --volumes-from=deis-logger deis/controller (code=exited, status=1/FAILURE)
  Process: 28875 ExecStartPre=/bin/sh -c docker inspect deis-controller >/dev/null && docker rm -f deis-controller || true (code=exited, status=0/SUCCESS)
  Process: 28868 ExecStartPre=/bin/sh -c docker history deis/controller >/dev/null || docker pull deis/controller:latest (code=exited, status=0/SUCCESS)
 Main PID: 28883 (code=exited, status=1/FAILURE)

Jun 06 15:45:49 sevilla docker[28883]: File "/usr/local/lib/python2.7/dist-packages/django/db/backends/__init__.py", line 115, in connect
Jun 06 15:45:49 sevilla docker[28883]: self.connection = self.get_new_connection(conn_params)
Jun 06 15:45:49 sevilla docker[28883]: File "/usr/local/lib/python2.7/dist-packages/django/db/backends/postgresql_psycopg2/base.py", line 114, in get_new_connection
Jun 06 15:45:49 sevilla docker[28883]: return Database.connect(**conn_params)
Jun 06 15:45:49 sevilla docker[28883]: File "/usr/local/lib/python2.7/dist-packages/psycopg2/__init__.py", line 164, in connect
Jun 06 15:45:49 sevilla docker[28883]: conn = _connect(dsn, connection_factory=connection_factory, async=async)
Jun 06 15:45:49 sevilla docker[28883]: django.db.utils.OperationalError: FATAL:  password authentication failed for user "deis"
Jun 06 15:45:49 sevilla systemd[1]: deis-controller.service: main process exited, code=exited, status=1/FAILURE
Jun 06 15:45:50 sevilla docker[29722]: deis-controller
Jun 06 15:45:50 sevilla systemd[1]: Unit deis-controller.service entered failed state.
```

**Comment by sgoings** [comment]
I currently have this issue on Azure with 1.10.

```
core@sgoings-test-deis-coreos-2 ~ $ etcdctl ls /deis/database/
/deis/database/adminPass
/deis/database/user
/deis/database/bucketName
/deis/database/initId
/deis/database/port
/deis/database/engine
/deis/database/adminUser
/deis/database/password
/deis/database/name
/deis/database/host
```

And running the following in the database container gives:

```
/app # psql -h 10.0.0.8 -p 5432 -U postgres -W
Password for user postgres:
psql: FATAL:  password authentication failed for user "postgres"
/app # psql -h 10.0.0.8 -p 5432 -U deis -W
Password for user deis:
psql: FATAL:  password authentication failed for user "deis"
```

**Comment by sgoings** [comment]
See also #2821

**Comment by sgoings** [comment]
Even after scaling each node to an A4 in Azure (and restarting the platform)... I still had the same behavior:

- etcd nodes report unhealthy fairly regularly (multiple times per node per minute)
- controller can't connect to database because `password authentication failed for user "deis"`
- builder can't start due to controller not being up
- etcd cluster-health loop shows that nodes are healthy -> not healthy -> healthy.

**Comment by mboersma** [comment]
I think this was (re-)fixed with [Deis v1.10.1](https://github.com/deis/deis/releases/tag/v1.10.1).
